plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -174,15 +169,15 @@ public void writeFailedPluginsToFile() {
 
         try (FileWriter fileWriter = new FileWriter("failedplugins.txt")) {
             if (failedPlugins.size() > 0) {
-                System.out.println("Some plugins failed to download: ");
+                LOGGER.log(Level.WARNING, "Some plugins failed to download: ");
                 for (Plugin plugin : failedPlugins) {
```

Comment:
Pretty much you can use lambdas to build the string:
`String failedPluginNames = failedPlugins.stream().map( p => p.getName()).collect(Collectors.joining(","));`

It can be multiple lines for clarity.